### PR TITLE
Epmlabsbrn 671 First realization API for User Statistic

### PR DESCRIPTION
**Pull request by nikolai4lazarev**
EPMLABSBRN-671 https://jira.epam.com/jira/browse/EPMLABSBRN-671

Updating api's contract

### CODE REVIEW

File: api-contract/api.raml [hunk 1]
```diff
@@ -34,7 +34,86 @@ mediaType:
...
+          type: Int,
+          default: current year,
+          example: 2021
+      response:
+        200:
+          body:
+            example: {
+              data: [
+                15:00: {
+                  seriesName: "name",
```
Comment (lifeart) [inline api-contract/api.raml:61]: we don't need time key here, it should be property on object

`data: [ { id, seriesName, level, spendTime, time } ]`

File: api-contract/api.raml [hunk 2]
```diff
@@ -34,7 +34,86 @@ mediaType:
...
+                  spentTime: 5,
+                  tasksCount: 4,
+                  wrongAnswers: 1,
+                  repetition: 1
+                }
+              ],
+              errors: [],
+              meta: []
+            }
+  /month:
```
Comment (lifeart) [inline api-contract/api.raml:74]: wondering, could we have one endpoint for all this needs?

object may include year, month

and api may look like: /statistics/study?from=ISO_DATE&to=ISO_DATE

File: api-contract/api.raml [hunk 3]
```diff
@@ -34,6 +34,59 @@ mediaType:
               errors: [],
               meta: []
             }
+  /study/week:
```
Comment (lifeart) [inline api-contract/api.raml:37]: could we make it work with custom userId for admin users? (to show stats per user/users in admin-panel)
cc @KomlevDm

Comment (nikolai4lazarev) [reply api-contract/api.raml:37]: Please, take a look at the admin section in the file. There is such of this logic already

**Review by lifeart — APPROVED** [review]

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -79,6 +79,7 @@ brn.pictureWithWord.file.default.path=pictures/withWord/%s.jpg
 brn.picture.file.default.path=pictures/%s.jpg
 brn.picture.theme.path=pictures/theme/%s.svg
 brn.dataFormatNumLines=5
+brn.statistic.goal.day=20
```
Comment (goopnigoop) [inline src/main/resources/application.properties:82]: is it crested for the future? I can not find a place where this one is used.

**Comment by goopnigoop** [comment]
@nikolai4lazarev like this one!

File: src/main/resources/application.properties (hunk 4 above)
Comment (nikolai4lazarev) [reply src/main/resources/application.properties:82]: Will remove it

File: api-contract/api.raml [hunk 5]
```diff
@@ -458,24 +511,84 @@ mediaType:
...
+      get:
+        description: get user's weekly statistic for period
+        queryParameters:
+          userId:
+            required: true
+            type: long
+            example: 1
+          from:
+            required: true
+            type: LocaDate
```
Comment (ElenaSpb) [inline api-contract/api.raml:542]: LocalDate?

File: api-contract/api.raml [hunk 6]
```diff
@@ -458,24 +511,84 @@ mediaType:
...
+            example: 2000-11-23
+          to:
+            required: true
+            type: LocaDate
```
Comment (ElenaSpb) [inline api-contract/api.raml:546]: LocalDate?

File: src/test/kotlin/com/epam/brn/integration/UserStatisticControllerIT.kt [hunk 7]
```diff
@@ -57,10 +85,72 @@ class UserStatisticControllerIT : BaseIT() {
...
-        Assertions.assertEquals(1, resultStatistic.first().totalExercises)
-        Assertions.assertEquals(1, resultStatistic.first().completedExercises)
+        assertEquals(1, resultStatistic.first().totalExercises)
+        assertEquals(1, resultStatistic.first().completedExercises)
+
+        assertEquals(0, resultStatistic[1].completedExercises)
+        assertEquals(1, resultStatistic[1].totalExercises)
+    }
+
+    @Test
```
Comment (ElenaSpb) [inline src/test/kotlin/com/epam/brn/integration/UserStatisticControllerIT.kt:95]: please follow in all tests GIVEN / WHEN /THEN approach to increase tests readability 
as described here https://github.com/Brain-up/brn/wiki/Coding-Standards

File: api-contract/api.raml [hunk 8]
```diff
@@ -34,6 +34,59 @@ mediaType:
...
+          type: date
+          example: 2000-11-23
+      response:
+        200:
+          body:
+            example: {
+              data: [
+                {
+                  date: 2020-11
+                  exercisingTime: 3000
```
Comment (KomlevDm) [inline api-contract/api.raml:82]: It is not clear what this dimension is.
For example like this: exercisingTime -> **exercisingTimeInSec**

File: api-contract/api.raml [hunk 9]
```diff
@@ -34,6 +34,59 @@ mediaType:
...
+                  progress: GREAT
```
Comment (KomlevDm) [inline api-contract/api.raml:83]: No information about the number of days with exercises. Please add.

image

File: api-contract/api.raml (hunk 8 above)
Comment (KomlevDm) [reply api-contract/api.raml:82]: And also in another such dtos

**Review by KomlevDm — APPROVED** [review]

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**